libs/agno/tests/unit/os/test_duplicate_ids.py

```diff
@@ -0,0 +1,165 @@
+        AgentOS(teams=[team1, team2], telemetry=False)
+
+    assert "Duplicate IDs found in AgentOS" in str(exc_info.value)
+    assert "same-team-id" in str(exc_info.value)
+
+
+def test_duplicate_team_ids_from_same_name_raises_error():
+    """Test that teams with the same name (generating same ID) raise ValueError."""
+    agent1 = Agent(name="Agent 1", telemetry=False)
+    agent2 = Agent(name="Agent 2", telemetry=False)
+
+    team1 = Team(name="My Team", members=[agent1])
+    team2 = Team(name="My Team", members=[agent2])
+
+    with pytest.raises(ValueError) as exc_info:
+        AgentOS(teams=[team1, team2], telemetry=False)
+
+    assert "Duplicate IDs found in AgentOS" in str(exc_info.value)
+
+
+def test_unique_team_ids_work_correctly():
+    """Test that unique team IDs are accepted."""
+    agent1 = Agent(name="Agent 1", telemetry=False)
+    agent2 = Agent(name="Agent 2", telemetry=False)
+
+    team1 = Team(name="Team 1", id="team-1", members=[agent1])
+    team2 = Team(name="Team 2", id="team-2", members=[agent2])
+
+    os = AgentOS(teams=[team1, team2], telemetry=False)
+    assert len(os.teams) == 2
+
+
+def test_duplicate_workflow_ids_with_explicit_ids_raises_error():
+    """Test that duplicate explicit workflow IDs raise ValueError."""
+    workflow1 = Workflow(name="Workflow 1", id="same-workflow-id")
+    workflow2 = Workflow(name="Workflow 2", id="same-workflow-id")
+
+    with pytest.raises(ValueError) as exc_info:
+        AgentOS(workflows=[workflow1, workflow2], telemetry=False)
+
+    assert "Duplicate IDs found in AgentOS" in str(exc_info.value)
+    assert "same-workflow-id" in str(exc_info.value)
+
+
+def test_duplicate_workflow_ids_from_same_name_raises_error():
+    """Test that workflows with the same name (generating same ID) raise ValueError."""
+    workflow1 = Workflow(name="My Workflow")
+    workflow2 = Workflow(name="My Workflow")
+
+    with pytest.raises(ValueError) as exc_info:
+        AgentOS(workflows=[workflow1, workflow2], telemetry=False)
+
+    assert "Duplicate IDs found in AgentOS" in str(exc_info.value)
+
+
+def test_unique_workflow_ids_work_correctly():
+    """Test that unique workflow IDs are accepted."""
+    workflow1 = Workflow(name="Workflow 1", id="workflow-1")
+    workflow2 = Workflow(name="Workflow 2", id="workflow-2")
+
+    os = AgentOS(workflows=[workflow1, workflow2], telemetry=False)
+    assert len(os.workflows) == 2
+
+
+# Mixed component tests
+
+
+def test_mixed_components_with_unique_ids():
+    """Test that mixed components with unique IDs work correctly."""
+    agent = Agent(name="Agent", id="agent-id", telemetry=False)
+    team = Team(name="Team", id="team-id", members=[Agent(name="Team Agent", telemetry=False)])
+    workflow = Workflow(name="Workflow", id="workflow-id")
+
+    os = AgentOS(agents=[agent], teams=[team], workflows=[workflow], telemetry=False)
+    assert len(os.agents) == 1
+    assert len(os.teams) == 1
+    assert len(os.workflows) == 1
```

Comment:
Let's add one like this to check that entities of different type with same ID dont raise